src/lib/ui/next/customNextPage.js

```diff
@@ -1,38 +1,96 @@
+
+  const [transactionLink, setTransactionLink] = useState<string | null>(null);
+  const [contractState, setContractState] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  // fetch the zkapp state when the page loads
   useEffect(() => {
     (async () => {
-      const { Mina, PublicKey } = await import('o1js');
-      const { Add } = await import('../../contracts/build/src/');
+      Mina.setActiveInstance(Mina.Network('https://api.minascan.io/node/devnet/v1/graphql'));
+      await fetchAccount({publicKey: zkAppAddress});
+      const num = zkApp.current.num.get();
+      setContractState(num.toString());
+      setLoading(false);
+    })();
+  }, []);
+
+  const updateZkApp = useCallback(async () => {
+    setTransactionLink(null);
+    setLoading(true);
+
+    try {
+      // Retrieve Mina provider injected by browser extension wallet
+      const mina = (window as any).mina;
+      const walletKey: string = (await mina.requestAccounts())[0];
+      console.log("Connected wallet address: " + walletKey);
+      await fetchAccount({publicKey: PublicKey.fromBase58(walletKey)});
 
-      // Update this to use the address (public key) for your zkApp account.
-      // To try it out, you can try this address for an example "Add" smart contract that we've deployed to
-      // Testnet B62qnTDEeYtBHBePA4yhCt4TCgDtA4L2CGvK7PirbJyX4pKH8bmtWe5.
-      const zkAppAddress = '';
-      // This should be removed once the zkAppAddress is updated.
-      if (!zkAppAddress) {
-        console.error(
-          'The following error is caused because the zkAppAddress has an empty string as the public key. Update the zkAppAddress with the public key for your zkApp account, or try this address for an example "Add" smart contract that we deployed to Testnet: B62qnTDEeYtBHBePA4yhCt4TCgDtA4L2CGvK7PirbJyX4pKH8bmtWe5'
-        );
+      // Compile the contract so that o1js has the proving key required to execute contract calls
+      console.log("Compiling Add contract to generate proving and verification keys");
+      await Add.compile();
+
+      // Execute a transaction locally on the browser
+      const transaction = await Mina.transaction(async () => {
+        console.log("Executing Add.update() locally");
+        await zkApp.current.update();
+      });
+
+      // Prove execution of the contract using the proving key
+      console.log("Proving execution of Add.update()");
+      await transaction.prove();
+
+      // Broadcast the transaction to the Mina network
+      console.log("Broadcasting proof of execution to the Mina network");
+      const {hash} = await mina.sendTransaction({transaction: transaction.toJSON()});
+
+      // display the link to the transaction
+      const transactionLink = "https://minascan.io/devnet/tx/" + hash;
+      setTransactionLink(transactionLink);
+    } catch (e: any) {
```

Comment:
Even though it's more steps, I find that `[Connect Wallet]` is a nice experience compared to the wallet auto-connecting and failing if it's not available.

Reply:
Are you thinking we have a "Connect Wallet" button in addition to the update button? Right now clicking the update button will trigger the connection attempt but nothing happens on page load

Reply:
Ah disregard my comment.  I misread where the ifs are!